### PR TITLE
Source hector, sip and pyqt from cms-externals.

### PR DESCRIPTION
**Pull request by ghost**
Hector: https://github.com/cms-externals/hector/commit/365361cb7771593c2ff0ad91aa842858d44ded6d
Sip: https://github.com/cms-externals/sip/commit/51f6c53b52d0330ba46ecfd772a0a03db18ec75f
PyQt: https://github.com/cms-externals/pyqt/commit/f909ea86706cd38b2d8b14d03205e896104b3c9b

Cleanup of patch files now included in cms-externals. 
+1
Tested compilation up to toolfile of each package.

### CODE REVIEW

File: hector.spec [hunk 1]
```diff
@@ -17,10 +14,9 @@ Patch2: hector-1.3.4-fix-narrowing-conversion
 %endif
 
 %prep
-%setup -q -n %{rname}_%{realversion}
-%patch0 -p2
-%patch1 -p1
-%patch2 -p1
+%setup -q -n %{n}.%{realversion}
+mkdir -p obj
```
Comment (ktf) [inline hector.spec]: For the record:

```
mkdir -p obj lib
```

should be preferred.

File: sip.spec [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+%define tag 51f6c53b52d0330ba46ecfd772a0a03db18ec75f
+%define branch cms/v%{realversion}
+%define github_user cms-externals
+Source: git+https://github.com/%github_user/%{n}.git?obj=%{branch}/%{tag}&export=%{n}.%{realversion}&output=/%{n}.%{realversion}-%{tag}.tgz
 Requires: python
+
 %prep
-%setup -n sip-%realversion 
-%patch0 -p1
+%setup -n %{n}.%{realversion}
```
Comment (ktf) [inline sip.spec]: Why having a . rather than a `-`?

**Comment by ktf** [comment]
Thanks. Can you clean up at least the second issue above (well, I guess the first one as well while you are at it)?

**Comment by ghost** [comment]
@ktf I updated the PR. All should be fine now.

**Pull request merged**